### PR TITLE
CCD-3760 CCD-3771 introduce CreatedBy

### PR DESCRIPTION
**Pull request by RebeccaBaker**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CCD-3760
https://tools.hmcts.net/jira/browse/CCD-3771

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/managecase/service/noc/ApplyNoCDecisionService.java [hunk 1]
```diff
@@ -120,14 +121,17 @@ private void applyDecision(CaseDetails caseDetails, JsonNode changeOrganisationR
         JsonNode organisationToRemoveNode = changeOrganisationRequestField.get(ORGANISATION_TO_REMOVE);
         Organisation organisationToAdd = objectMapper.convertValue(organisationToAddNode, Organisation.class);
         Organisation organisationToRemove = objectMapper.convertValue(organisationToRemoveNode, Organisation.class);
+        JsonNode createdBy = changeOrganisationRequestField.get(CREATED_BY);
 
         List<CaseUserRole> existingCaseAssignments = dataStoreRepository
             .getCaseAssignments(singletonList(caseDetails.getId()), null);
 
         if (organisationToAdd == null || isNullOrEmpty(organisationToAdd.getOrganisationID())) {
+            ((ObjectNode) orgPolicyNode).putIfAbsent("LastNoCRequestedBy", null);
```
Comment (danlysiak) [inline src/main/java/uk/gov/hmcts/reform/managecase/service/noc/ApplyNoCDecisionService.java:130]: This should probably sit in the `applyRemoveRepresentationDecision()` method

File: src/main/java/uk/gov/hmcts/reform/managecase/client/datastore/CaseDetails.java [hunk 2]
```diff
@@ -33,6 +33,7 @@
 public class CaseDetails {
 
     public static final String ORGANISATION_TO_ADD = "OrganisationToAdd";
+    public static final String CREATED_BY = "CreatedBy";
```
Comment (danlysiak) [inline src/main/java/uk/gov/hmcts/reform/managecase/client/datastore/CaseDetails.java:36]: We should probably add `LastNoCRequestedBy` as a constant here too.

File: src/main/java/uk/gov/hmcts/reform/managecase/service/noc/ApplyNoCDecisionService.java (hunk 1 above)
Comment (danlysiak) [reply src/main/java/uk/gov/hmcts/reform/managecase/service/noc/ApplyNoCDecisionService.java:130]: And can we check whether `putIfAbsent()` works in all scenarios - won't this only set to null if there's currently no value?
Might need another approach to always nullify it e.g.:
```suggestion
            ((ObjectNode) orgPolicyNode).set("LastNoCRequestedBy", objectMapper.nullNode());
```

File: src/main/java/uk/gov/hmcts/reform/managecase/service/noc/ApplyNoCDecisionService.java [hunk 3]
```diff
@@ -120,14 +121,17 @@ private void applyDecision(CaseDetails caseDetails, JsonNode changeOrganisationR
...
             applyRemoveRepresentationDecision(existingCaseAssignments, caseRoleId, orgPolicyNode, organisationToRemove,
                 caseDetails.getId());
         } else {
+            ((ObjectNode) orgPolicyNode).putIfAbsent("LastNoCRequestedBy", createdBy);
```
Comment (danlysiak) [inline src/main/java/uk/gov/hmcts/reform/managecase/service/noc/ApplyNoCDecisionService.java:134]: Similar comments to above - should probably sit in the `applyAddOrReplaceRepresentationDecision()` method, and `putIfAbsent()` presumably won't actually update it if there's an existing one (but we do want it to)?

File: src/main/java/uk/gov/hmcts/reform/managecase/client/datastore/CaseDetails.java (hunk 2 above)
Comment (RebeccaBaker) [reply src/main/java/uk/gov/hmcts/reform/managecase/client/datastore/CaseDetails.java:36]: done

File: src/main/java/uk/gov/hmcts/reform/managecase/service/noc/ApplyNoCDecisionService.java (hunk 1 above)
Comment (RebeccaBaker) [reply src/main/java/uk/gov/hmcts/reform/managecase/service/noc/ApplyNoCDecisionService.java]: done

**Review by danlysiak — APPROVED** [review]

**Pull request merged**